### PR TITLE
Use cilium as kube-proxy replacement

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/roadmap/issues/1072

### CODE REVIEW

File: helm/cluster-cloud-director/files/var/run/kubeadm/gs-update-kubeadm-yaml-skip-kube-proxy.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+readonly dir
+
+# TODO Change to giantswarm/yq # (https://github.com/giantswarm/retagger/pull/810)
```
Comment (kopiczko) [inline helm/cluster-cloud-director/files/var/run/kubeadm/gs-update-kubeadm-yaml-skip-kube-proxy.sh:10]: To self: sort before merging

File: helm/cluster-cloud-director/templates/_helpers.tpl [hunk 2]
```diff
@@ -95,17 +95,6 @@ use the cluster-apps-operator created secret <clusterName>-cluster-values as def
...
-- path: /run/kubeadm/gs-kube-proxy-config.yaml
-  permissions: "0600"
-  content: |
-    {{- .Files.Get "files/etc/gs-kube-proxy-config.yaml" | nindent 4 }}
-- path: /run/kubeadm/gs-kube-proxy-patch.sh
-  permissions: "0700"
-  content: |
-    {{- .Files.Get "files/etc/gs-kube-proxy-patch.sh" | nindent 4 }}
-{{- end -}}
-
```
Comment (kopiczko) [inline helm/cluster-cloud-director/templates/_helpers.tpl:108]: Moved to _kubeadm.tpl

File: helm/cluster-cloud-director/files/run/kubeadm/gs-update-kubeadm-yaml-skip-kube-proxy.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+fi
+
+# This will add the configuration below to "InitConfiguration":
+#
+#     skipPhases:
+#       - addon/kube-proxy
+#
+set -x
+sed -i 's/nodeRegistration:/skipPhases:\n  - addon\/kube-proxy\nnodeRegistration:/' "${dir}/kubeadm.yaml"
+{ set +x; } 2>/dev/null
```
Comment (kopiczko) [inline helm/cluster-cloud-director/files/run/kubeadm/gs-update-kubeadm-yaml-skip-kube-proxy.sh:22]: This file will be gone if we can update cluster-api-app and we have skipPhases available (check TODO in helm/cluster-cloud-director/templates/kubeadmcontrolplane.yaml)

File: helm/cluster-cloud-director/templates/_kubeadm.tpl [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  content: |
+    {{- .Files.Get "files/run/kubeadm/gs-update-kubeadm-yaml-skip-kube-proxy.sh" | nindent 4 }}
+- path: /run/kubeadm/gs-kube-proxy-config.yaml
+  permissions: "0600"
+  content: |
+    {{- .Files.Get "files/run/kubeadm/gs-kube-proxy-config.yaml" | nindent 4 }}
+- path: /run/kubeadm/gs-kube-proxy-patch.sh
+  permissions: "0700"
+  content: |
+    {{- .Files.Get "files/run/kubeadm/gs-kube-proxy-patch.sh" | nindent 4 }}
```
Comment (vxav) [inline helm/cluster-cloud-director/templates/_kubeadm.tpl:16]: Why do we still need `gs-kube-proxy-config.yaml` ?

**Review by vxav — APPROVED** [review]
lgtmiiwfy

File: helm/cluster-cloud-director/templates/_kubeadm.tpl (hunk 4 above)
Comment (kopiczko) [reply helm/cluster-cloud-director/templates/_kubeadm.tpl:16]: 🤦

**Pull request merged**